examples_test/config_test.go

```diff
@@ -0,0 +1,144 @@
+ HieraConfig is mutually exclusive to use of HieraRoot and HieraConfigFileName.
+
+ HieraRoot will default to the current working directory
+
+ HieraConfigFileName will default to "hiera.yaml"
+
+ If the HieraRoot or HieraConfig are relative paths, they will be considered relative to the current directory.
+
+ The HieraConfigFileName must be relative to the HieraRoot (or the current directory).
+*/
+
+// TestHelloWorld_yamlConfig uses the special "lookup_key" function ConfigLookupKey and HieraRoot.
+func TestHelloWorld_yamlConfig(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+	configOptions[hieraapi.HieraRoot] = types.WrapString(`testdata`)
+
+	// Initialize a Hiera session with the ConfigLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.ConfigLookupKey, configOptions, func(c px.Context) {
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+	})
+}
+
+// TestHelloWorld_explicitYamlConfig is similar to TestHelloWorld_yamlConfig but uses HieraConfig
+// option to explicitly define what file to use.
+func TestHelloWorld_explicitYamlConfig(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+	configOptions[hieraapi.HieraConfig] = types.WrapString(`testdata/hiera.yaml`)
+
+	// Initialize a Hiera session with the ConfigLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.ConfigLookupKey, configOptions, func(c px.Context) {
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+	})
+}
+
+// TestHelloWorld_explicitYamlConfigFile is similar to TestHelloWorld_yamlConfig but uses a combination of
+// HieraRoot and HieraConfigFileName option.
+func TestHelloWorld_explicitYamlConfigFile(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+	configOptions[hieraapi.HieraRoot] = types.WrapString(`testdata`)
+	configOptions[hieraapi.HieraConfigFileName] = types.WrapString(`special.yaml`)
+
+	// Initialize a Hiera session with the ConfigLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.ConfigLookupKey, configOptions, func(c px.Context) {
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml special data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+	})
+}
+
+// TestHelloWorld_yamlAndSemiHardWired uses the MuxLookupKey to inject two lookup_key functions. The ConfigLookupKey
+// that consults the yaml config and the sayHello.
```

Comment:
This provides a clue to MuxLookupKey - it seems to build an aggregate function having a hierarchy of looking in one than the other - assuming the example shows configured to have higher precedence maybe spell that out.